pkg/testmachinery/testdefinition/testdefinition.go

```diff
@@ -210,52 +213,63 @@ func (td *TestDefinition) AddSerialStdOutput() {
 }
 
 // AddConfig adds the config elements of different types (environment variable) to the TestDefinitions's template.
-func (td *TestDefinition) AddConfig(configs []*config.Element) {
-	for _, config := range configs {
-		switch config.Info.Type {
+func (td *TestDefinition) AddConfig(configs []*config.Element) error {
```

Comment:
Very long and nested function. I think it makes sense to introduce 2 new function which handle the addConfigAsFile and addConfigAsEnv

Reply:
Factored out to 2 function